src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -253,19 +267,101 @@ class DataFrameReader:
     """
 
     def __init__(self, session: "snowflake.snowpark.session.Session") -> None:
+        self._cur_options: dict[str, Any] = {}
         self._session = session
         self._user_schema: Optional[StructType] = None
-        self._cur_options: dict[str, Any] = {}
         self._file_path: Optional[str] = None
         self._file_type: Optional[str] = None
-        self._metadata_cols: Optional[Iterable[ColumnOrName]] = None
+        self._metadata_cols: Optional[Iterable[_MetadataColumn]] = None
         # Infer schema information
         self._infer_schema = False
         self._infer_schema_transformations: Optional[
             List["snowflake.snowpark.column.Column"]
         ] = None
         self._infer_schema_target_columns: Optional[List[str]] = None
 
+    def load(self, path: str, format: str = None, schema: StructType = None, **kwargs):
+        """
```

Comment:
Can you declare a `Literal` for supported file formats and use that here and in `def format` below.

https://peps.python.org/pep-0586/

Reply:
Good suggestion. Thanks